test/cpp/jit/test_custom_class_registrations.cpp

```diff
@@ -339,9 +339,10 @@ TORCH_LIBRARY(_TorchScriptTesting, m) {
           },
           [](std::vector<int64_t> state) { // __setstate__
             return c10::make_intrusive<Foo>(state[0], state[1]);
-          });
-  m.def(
-      "takes_foo(__torch__.torch.classes._TorchScriptTesting._Foo foo, Tensor x) -> Tensor");
+          })
+      .def_meta([](c10::intrusive_ptr<Foo> self) { // __get_metadata__
```

Comment:
nb: we could totally just ask the user to `.def("__get_metadata__")` right now, right?

Reply:
yeah, they can!